Automattic/pocket-casts-ios
# Cleanup: Removes ProfileIntroViewController

- Removes the `ProfileIntroViewController` class and xib
- Updates the PromotionViewController to use the new onboarding/upgrade flow

## To test
1. Launch the app
2. Sign out if you're signed in
3. In Terminal run `xcrun simctl openurl booted pktc://redeem/promo/test`
4. ✅ Verify you see the promotion view
5. ✅ Verify tapping create an account dismisses the view and opens the new login landing page
12. Run the command again in using an expired code or change the `promoStatus` in viewDidLoad to `.codeExpired`
13. Tap the upgrade button
14. ✅ Verify the view dismisses and you are brought to the new upgrade flow
6. Dismiss the view
7. Run the command in Terminal but with a valid promo code or change the `promoStatus` in viewDidLoad to `.signIn`
8. ✅ Verify you see a prompt to sign in
9. Tap next
10. Sign in or create an account
11. ✅ Verify you see the promotion redeemed message


## Checklist
